tests/Kros.AspNetCore.Tests/Extensions/DistributedCacheExtensionsShould.cs

```diff
@@ -0,0 +1,125 @@
+﻿using FluentAssertions;
+using Microsoft.Extensions.Caching.Distributed;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Kros.AspNetCore.Tests.Extensions
+{
+    public class DistributedCacheExtensionsShould
+    {
+        #region Test class
+
+        class Foo
+        {
+            public int Value { get; set; }
+        }
+
+        #endregion
+
+        [Fact]
+        public async Task SetValueToCache()
+        {
+            IDistributedCache cache = new MemoryDistributedCache();
+
+            await cache.SetAsync("2", 2, new DistributedCacheEntryOptions());
+
+            var value = await cache.GetAsync<int>("2");
+
+            value.Should().Be(2);
+        }
+
+        [Fact]
+        public async Task SetComplexValueToCache()
+        {
+            IDistributedCache cache = new MemoryDistributedCache();
+
+            await cache.SetAsync("foo", new Foo() { Value = 2 }, new DistributedCacheEntryOptions());
+
+            var value = await cache.GetAsync<Foo>("foo");
+
+            value.Value.Should().Be(2);
+        }
+
+        [Fact]
+        public async Task GetDefaultWhenKeyDoesNotExist()
+        {
+            IDistributedCache cache = new MemoryDistributedCache();
+
+            var foo = await cache.GetAsync<Foo>("foo");
+            foo.Should().BeNull();
+
+            var @int = await cache.GetAsync<int?>("int");
+            @int.Should().BeNull();
+
+            var @string = await cache.GetAsync<string>("string");
+            @string.Should().BeNull();
+        }
+
+        [Fact]
+        public async Task GetAndSetIfDoNotExistInCache()
+        {
+            IDistributedCache cache = new MemoryDistributedCache();
+            DistributedCacheEntryOptions options = new DistributedCacheEntryOptions();
+            var callCount = 0;
+            Func<Foo> func = () =>
+            {
+                callCount++;
+                return new Foo() { Value = 2 };
+            };
```

Comment:
If using local function above.

```suggestion
            }
```